front-end/src/components/Card/Card.js

```diff
@@ -1,23 +1,90 @@
+import { getUsers } from '../../redux/nodes/entities/users/selectors';
 import { StyledCard, StyledCardContent } from './styled';
 import { propTypes, defaultProps } from './props';
+import { Comment } from './Comment';
+import { GridContainer, GridItem } from '../Grid';
 import { Text } from '../Text';
+import { Button } from '../Button';
+import { CreateCommentForm } from '../Forms';
+
+export const Card = ({ post, ...rest }) => {
+  const dispatch = useDispatch();
+  const { userId } = useSelector(getData);
+  const { comments } = useSelector(getComments);
+  const { users } = useSelector(getUsers);
+  const [liked, setLiked] = useState(false);
+
+  useEffect(() => {
+    setLiked(post.likes.indexOf(userId) !== -1);
+  }, [post.likes]);
 
-export const Card = ({ children, ...rest }) => {
   return (
     <StyledCard {...rest}>
       <StyledCardContent>
-        <Text variant='h5' gutterBottom>
-          This is a title
-        </Text>
-        <Text variant='body2'>
-          Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam saepe,
-          alias nesciunt possimus cupiditate a suscipit eligendi recusandae
-          harum pariatur numquam distinctio, quisquam sequi eveniet nostrum
-          libero accusamus mollitia. Culpa saepe id, eaque a sed distinctio
-          maiores nihil itaque accusantium eveniet eligendi assumenda
-          consequuntur sequi perferendis omnis, quo temporibus? Aut!
-        </Text>
+        <GridContainer>
+          <GridItem xs={12}>
+            <Text variant='h5' style={{ fontWeight: 'bold' }} gutterBottom>
+              {users[post.user].name}
+            </Text>
+          </GridItem>
+
+          <GridItem xs={12}>
+            <Text
+              variant='body2'
+              color='textSecondary'
+              gutterBottom
+              style={{ display: 'flex', alignItems: 'center', gap: '5px' }}
+            >
+              <WatchLater style={{ fontSize: 'inherit' }} />
+              {`${moment(post.timestamp).fromNow()}`}
+            </Text>
+          </GridItem>
+
+          <GridItem xs={12}>
+            <Text variant='body1' gutterBottom>
+              {post.content}
+            </Text>
+          </GridItem>
+
+          <GridItem xs={12}>
+            <Button
+              variant='outlined'
+              startIcon={liked ? <ThumbUpAlt /> : <ThumbUpAltOutlined />}
+              onClick={() =>
+                dispatch(liked ? unlikePost(post._id) : likePost(post._id))
+              }
+              style={{ marginBottom: '10px' }}
+            >
+              {`${post.likes.length} likes`}
+            </Button>
+          </GridItem>
+
+          <GridItem xs={12}>
+            <Text variant='h6' gutterBottom>
+              Comments
+            </Text>
+          </GridItem>
+
+          <GridItem xs={12}>
+            <CreateCommentForm postId={post._id} />
+          </GridItem>
+
+          {post.comments.map(commentId => (
```

Comment:
Install lodash library and use map from that library.

Reply:
Now using map function from lodash library.